dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/ToolTest.java

```diff
@@ -3441,6 +3643,272 @@ public void testAddPersonToListToolResponse() throws Exception {
         JSONAssert.assertEquals(expectedResponseString, response, true);
     }
 
+    @Test
+    public void testAddPersonToListToolResponseWithMetaRequest() throws Exception {
```

Comment:
This test is good because it tests a complex scenario and shows that the implementation works in non-trivial cases.

However, it would be good to have a really simple test for this scenario as well which is easier to read and debug.